python/ray/experimental/workflow/workflow_storage.py

```diff
@@ -389,6 +393,21 @@ def load_workflow_meta(self) -> Optional[WorkflowMetaData]:
         except KeyNotFoundError:
             return None
 
+    def get_named_step_id(self, name: str) -> Optional[str]:
+        """Get the step id from the step name
+
+        Args:
+            name: the name of the step
+
+        Returns:
+            The step id associated with the name. Return None if it doesn't
```

Comment:
What if there are duplicate steps with the same name?

Reply:
It's overwriting semantic right now. We might extend it to a mapping: name -> [steps]